packages/alert/src/Alert.tsx

```diff
@@ -69,9 +69,34 @@ export default function Alert({
   variant = 'success',
   ...other
 }: AlertProps): React.ReactElement<AlertProps> {
+  const [show, setShow] = useState(true)
   const Icon = icons[variant]
+
+  useEffect(() => {
+    let timer: NodeJS.Timeout
+    if (type === 'toast') {
+      timer = setTimeout(hideAlert, 5 * 1000)
+    }
+
+    return () => {
+      if (timer) clearTimeout(timer)
+    }
+  }, [type])
+
+  /* istanbul ignore next */
+  const hideAlert = () => {
+    setShow(false)
+    onClose?.()
+  }
+
   return (
-    <AlertWrapper role="alert" variant={variant} type={type} {...other}>
+    <AlertWrapper
+      role="alert"
+      variant={variant}
+      type={type}
+      className={show ? 'show' : 'hide'}
```

Comment:
Also, let's not use class names to control the behavior; we use styled-components, so this can be a prop we pass down.